robocop/checkers/misc.py

```diff
@@ -525,15 +564,47 @@ def check_adjacent_ifs(self, node):
                 previous_if = None
         self.generic_visit(node)
 
-    @staticmethod
-    def compare_conditions(if_node, other_if_node):
+    def compare_conditions(self, if_node, other_if_node):
+        if not self.compare_assign(if_node, other_if_node):
+            return False
         while if_node is not None and other_if_node is not None:
             if if_node.condition != other_if_node.condition:
                 return False
             if_node = if_node.orelse
             other_if_node = other_if_node.orelse
         return if_node is None and other_if_node is None
 
+    @staticmethod
+    def normalize_var_name(name):
+        return name.lower().replace("_", "").replace(" ", "").replace("=", "")
+
+    def compare_assign(self, if_node, other_if_node):
+        assign_1 = getattr(if_node, "assign", None)
+        assign_2 = getattr(other_if_node, "assign", None)
+        if assign_1 is None or assign_2 is None:
+            return all(assign is None for assign in (assign_1, assign_2))
+        if len(assign_1) != len(assign_2):
+            return False
+        for var1, var2 in zip(assign_1, assign_2):
+            if self.normalize_var_name(var1) != self.normalize_var_name(var2):
+                return False
+        return True
+
+    @staticmethod
+    def tokens_length(tokens):
+        return sum(len(token.value) for token in tokens)
+
+    def check_if_should_be_inline(self, node):
```

Comment:
I think I would rephrase the method name to `check_whether_if_should_be_inline` (or maybe even `check_whether_if_condition_should_be_inline`) because currently it is a bit confusing.

Reply:
Going for ``check_whether_if_should_be_inline`` (since it's whole IF that could be inline IF, not only condition)